azurerm/internal/services/netapp/resource_arm_netapp_volume.go

```diff
@@ -114,22 +129,25 @@ func resourceArmNetAppVolume() *schema.Resource {
 								ValidateFunc: validate.CIDR,
 							},
 						},
-						"cifs_enabled": {
-							Type:     schema.TypeBool,
-							Required: true,
-						},
-						"nfsv3_enabled": {
-							Type:     schema.TypeBool,
-							Required: true,
-						},
-						"nfsv4_enabled": {
-							Type:     schema.TypeBool,
+
+						"protocols_enabled": {
+							Type:     schema.TypeList,
 							Required: true,
```

Comment:
And then this can become optional, and we use the _enabled functions, unless this is set (we can fix this all up in 2.0)